src/test/util/setup_common.cpp

```diff
@@ -77,9 +77,9 @@ std::ostream& operator<<(std::ostream& os, const uint256& num)
 
 BasicTestingSetup::BasicTestingSetup(const std::string& chainName, const std::vector<const char*>& extra_args)
     : m_path_root{fs::temp_directory_path() / "test_common_" PACKAGE_NAME / g_insecure_rand_ctx_temp_path.rand256().ToString()},
-      m_args{}
+      m_args{gArgs}
```

Comment:
In commit "test/setup: Unify m_args and gArgs" (7ede150299ea46d38a0797910c95028e7e948e0d)

I think it would be better to drop this change, and let `BasicTestingSetup::m_args` keep being a variable instead of a `gArgs` reference. The reason the `BasicTestingSetup::m_args` member exists is to provide a way for tests to stop using a shared `gArgs` instance and let each test have it's own standalone ArgsManager. This way tests can do `m_args.ForceSetArg` and other operations without affecting global state and potentially breaking each other.

The other change below which stops using `gArgs` would be good to keep if it still works:

```diff
-    m_node.args = &gArgs;
+    m_node.args = &m_args;
```

If it doesn't work now, it should work later when @kiminuo gets around to removing more `gArgs` uses in other code https://github.com/bitcoin/bitcoin/pull/21244#discussion_r610080299